app/models/asset.rb

```diff
@@ -94,9 +95,16 @@ def dimensions_for(style)
   end
 
   def reprocess_for(style)
+    return if processing?
+
     if file.styles.has_key?(style) && ! file_dimensions.has_key?(style.to_s)
-      file.reprocess!(style)
-      save
+      begin
+        set(:processing, true)
```

Comment:
Are we worried about race conditions? Given one of the main situations we're trying to deal with here is when there are multiple simultaneous requests processing the file, maybe we should be.